main.go

```diff
@@ -1,33 +1,35 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2020 Bonc.com.cn, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+
+/**
+ * @file    main_test.go.go
+ * @author  qihouliang(qihouliang@bonc.com.cn)
+ * @date    2023/4/11 22:26
+ * @brief
+ */
+
 package main
 
 import (
-	"context"
-	"os"
-	"os/signal"
-	"syscall"
-
+	"fmt"
 	"github.com/appleboy/CodeGPT/cmd"
+	"github.com/appleboy/CodeGPT/webhook"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
 )
 
-func withContextFunc(ctx context.Context, f func()) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
-		defer signal.Stop(c)
-
-		select {
-		case <-ctx.Done():
-		case <-c:
-			cancel()
-			f()
-		}
-	}()
-
-	return ctx
-}
-
 func main() {
-	ctx := withContextFunc(context.Background(), func() {})
-	cmd.Execute(ctx)
+	log.SetFlags(log.Lshortfile | log.LstdFlags)
+	cmd.InitConfig()
+	http.HandleFunc("/payload", webhook.HandleWebhook)
+	ip := viper.GetString("server.ip")
+	port := viper.GetString("server.port")
+	var address = fmt.Sprintf("%s:%s", ip, port)
+	fmt.Println("Listening for GitHub Webhooks on", address)
+	http.ListenAndServe(address, nil)
+
 }
```

Comment:
Overall, the code patch looks relatively simple and straightforward. However, there are a few potential issues and areas for improvement:

1. Error handling: The code does not have much error handling in place. For example, if there is an error starting the HTTP server, the program will simply panic. It would be better to handle errors more gracefully and provide useful feedback to the user.

2. Security: The code does not appear to have any security measures in place, such as authentication or authorization, for handling incoming webhooks. This could potentially lead to security vulnerabilities if the webhook endpoint is exposed publicly.

3. Code organization: The code could benefit from better organization and separation of concerns. For example, the `main()` function is responsible for initializing configuration, starting the HTTP server, and handling webhooks. It would be better to break these responsibilities into separate functions or packages.

4. Testing: The code does not have any tests in place, which could make it difficult to catch regressions or ensure correct behavior as the codebase grows.

5. Documentation: The code could benefit from better documentation, both in terms of code comments and external documentation for users who want to set up and use the webhook.